drivers/spi/spi_nrfx_spim.c

```diff
@@ -184,8 +185,11 @@ static int configure(const struct device *dev,
 	config.mode      = get_nrf_spim_mode(spi_cfg->operation);
 	config.bit_order = get_nrf_spim_bit_order(spi_cfg->operation);
 
-	nrfy_gpio_pin_write(nrfy_spim_sck_pin_get(dev_config->spim.p_reg),
-			    spi_cfg->operation & SPI_MODE_CPOL ? 1 : 0);
+	nrfy_sck_pin = nrfy_spim_sck_pin_get(dev_config->spim.p_reg);
+	if (nrfy_sck_pin >= 0) {
```

Comment:
```suggestion
	if (nrfy_sck_pin != NRF_SPIM_PIN_NOT_CONNECTED) {
```

Reply:
This may not always work. The signal is disconnected when bit 31 is set to 1, but the register value doesn't have to be 0xFFFFFFFF. I'd rather suggest to use:
```suggestion
	if (FIELD_GET(nrfy_sck_pin, SPIM_PSEL_SCK_CONNECT_Msk) ==
	    SPIM_PSEL_SCK_CONNECT_Connected) {
```